### PR TITLE
Eslint fixes

### PR DESCRIPTION
**Pull request by OlegMoshkovich**

### CODE REVIEW

File: src/Components/AboutPanel.jsx [hunk 1]
```diff
@@ -1,117 +1,140 @@
...
-
-export function AboutControl ({offsetTop}) {
-  const [open, setOpen]=React.useState(true);
-  const classes = useStyles();
+/**
+ * Button to toggle About panel on and off
+ * @param {string} offset tree element
+ * @return {Component} componet
+ */
+export function AboutControl({offsetTop}) {
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/Components/AboutPanel.jsx:12]: As discussed, please do make export default.

File: src/utils/Ifc.js [hunk 2]
```diff
@@ -67,14 +95,24 @@ export function reifyName(element, viewer) {
   return prettyType(element, viewer) + ''
 }
 
-
+/**
+ * Elements description.
```
Comment (pablo-mayrgundter) [inline src/utils/Ifc.js:99]:
```
/**
 * Get the 'Description' property of the given element.  The string will also be decoded for non-ascii characters.
 * @param {Object} element IFC element.
 * @return {function|string} The element's description property.
 */

File: src/utils/IfcMock.test.js [hunk 3]
```diff
@@ -40,12 +48,23 @@ export class MockViewer {
...
   }
 }
 
-
+/**
+* Get the IFC type.
+* @param {string} label
+* @param {Number} id
+* @return {Object} returns property object
+*/
```
Comment (pablo-mayrgundter) [inline src/utils/IfcMock.test.js:67]:
```
/**
 * Create a mock element with the given label.
 * @param {string} label
 * @param {Number} id Express ID for the element.
 * @return {Object} Mock IFC element
 */
```

File: src/utils/IfcMock.test.js [hunk 4]
```diff
@@ -40,12 +48,23 @@ export class MockViewer {
     }
   }
 
+  /**
+  * Get the IFC type.
+  * @param {string} modelId
+  * @param {string} id
+  * @return {Object} returns property object
+  */
```
Comment (pablo-mayrgundter) [inline src/utils/IfcMock.test.js:56]:
```
/**
 * Get the IFC type.
 * @param {Number} modelId
 * @param {Number} id Express ID
 * @return {Object} Properties object
 */
```

File: src/utils/debug.js [hunk 5]
```diff
@@ -1,4 +1,9 @@
 const DEBUG_LEVEL = 4
+/**
```
Comment (pablo-mayrgundter) [inline src/utils/debug.js:2]: Indent

File: src/utils/debug.js [hunk 6]
```diff
@@ -1,4 +1,9 @@
...
+* Create debug statement.
+* @param {string} level
```
Comment (pablo-mayrgundter) [inline src/utils/debug.js:4]: @param {Number}

File: src/utils/gtag.js [hunk 7]
```diff
@@ -1,7 +1,10 @@
 window.dataLayer = window['dataLayer'] || []
+
+
+/** Create google analytics tag. */
 export default function gtag() {
   const dataLayer = window.dataLayer
-  dataLayer.push(arguments)
```
Comment (pablo-mayrgundter) [inline src/utils/gtag.js:4]: I think this is valid per:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments
.. maybe add this link as a comment and then eslint disble next line

File: src/utils/objects.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+/**
+  * TODO: pablo to confirm
+  * Delete all properties defined in the given object.
+  * @param {Object} obj index of the element in the set
+  */
```
Comment (pablo-mayrgundter) [inline src/utils/objects.js:5]:
```
/**
 * Delete all properties defined in the given object.
 * @param {Object} obj The object whose properties to delete.
 */

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

File: src/utils/gtag.js (hunk 7 above)
Comment (OlegMoshkovich) [reply src/utils/gtag.js:4]: not sure about indentation

File: src/Components/AboutPanel.jsx [hunk 9]
```diff
@@ -1,117 +1,138 @@
...
 import About from '../assets/3D/attention.svg'
 
-
-export function AboutControl ({offsetTop}) {
-  const [open, setOpen]=React.useState(true);
-  const classes = useStyles();
+/**
+ * Button to toggle About panel on and off
+ * @param {Number} offsetTop offset tree element
+ * @return {Component} componet
```
Comment (pablo-mayrgundter) [inline src/Components/AboutPanel.jsx:10]: @return {Object} The AboutControl react component.

File: src/utils/TreeUtils.js [hunk 10]
```diff
@@ -22,7 +29,12 @@ function setupLookupAndParentLinks(rootElt, elementsById) {
   })
 }
 
-
+/**
+ * generate URL address to the element
+ * @param {Object} elt IFC element.
+ * @param {string} getNameCb Instance of.
+ * @return {string} The sum of the two numbers.
+ */
```
Comment (pablo-mayrgundter) [inline src/utils/TreeUtils.js:37]:
```
/**
 * Generate a URL address fragment for the element.
 * @param {Object} elt IFC element.
 * @param {function} getNameCb Instance of.
 * @return {string} The URL path fragment for the element.
 */
```

Also, let's inline the exports for this function and the rest.

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

File: src/Containers/SearchIndex.js [hunk 11]
```diff
@@ -1,12 +1,15 @@
...
 import * as Ifc from '../utils/Ifc.js'
-
-
-const debug = 0
+import {deleteProperties} from '../utils/objects'
 
 
 /** TODO(pablo): maybe refactor into {IfcSearchIndex extends SearchIndex}. */
 export default class SearchIndex {
+/**
```
Comment (pablo-mayrgundter) [inline src/Containers/SearchIndex.js:9]: Indent

File: src/Containers/SearchIndex.js [hunk 12]
```diff
@@ -57,12 +63,21 @@ export default class SearchIndex {
...
   tokenize(str) {
     return new Set(str.match(/(\w+)/g))
   }
 
-
+  /**
+   * Create index set of found results
+   * @param {Object} index
+   * @param {string} key
+   * @return {str} set
```
Comment (pablo-mayrgundter) [inline src/Containers/SearchIndex.js:79]: @return {Object} The index set.

File: src/Containers/SearchIndex.js [hunk 13]
```diff
@@ -71,30 +86,43 @@ export default class SearchIndex {
...
-    }
-    for (const key in this.eltsByName) {
-      delete this.eltsByName[key]
-    }
+    deleteProperties(this.eltsByType)
+    deleteProperties(this.eltsByName)
   }
 
-
+  /**
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/Containers/SearchIndex.js:118]: My bad, @return {Array} resultIDs

File: src/utils/Ifc.js [hunk 14]
```diff
@@ -1,17 +1,31 @@
 import {stoi} from './strings'
 
-
+/**
+ * Determing if the given object is a type/value pair.
+ * @param {Object} obj IFC element.
+ * @return {boolean} The sum of the two numbers.
+ */
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/utils/Ifc.js:8]: I don't think that needs to be line wrapped:
```
/**
 * Check whether both type and value properties are defined and non-null on the object.
 * @param {Object} obj IFC element.
 * @return {boolean} True if and only if the both type and value properties are defined on the object.
 */
 ```

File: src/utils/TreeUtils.js [hunk 15]
```diff
@@ -1,4 +1,9 @@
-function visitTree(elt, observeCb) {
+/**
+ * Recursively visit nodes if the node has children
+ * @param {Elt} elt tree element
```
Comment (pablo-mayrgundter) [inline src/utils/TreeUtils.js:3]: @param {Object} ...

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

File: src/Containers/CadView.jsx [hunk 16]
```diff
@@ -220,7 +220,9 @@ export default function CadView({installPrefix, appPrefix, pathPrefix}) {
 
   /** Unpick active scene elts and remove clip planes. */
   function unSelectItems() {
-    viewer.unpickIfcItems()
+    console.log('in the unselect item',viewer)
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:223]: remove

**Review by pablo-mayrgundter — COMMENTED** [review]
Remove the one debug log then I think you have to merge in the hotfix to main from earlier and let's click around testing a lot.

File: src/BaseRoutes.jsx [hunk 17]
```diff
@@ -46,6 +46,7 @@ export default function BaseRoutes({testElt = null}) {
       debug().log('BaseRoutes: forwarding to: ', installPrefix + '/share')
       navigate(installPrefix + '/share')
     }
+  // eslint-disable-next-line
```
Comment (pablo-mayrgundter) [inline src/BaseRoutes.jsx:49]: Needs the specific error

File: src/Components/GuidePanel.jsx [hunk 18]
```diff
@@ -1,9 +1,61 @@
 import React from 'react'
 import {makeStyles} from '@mui/styles'
```
Comment (pablo-mayrgundter) [inline src/Components/GuidePanel.jsx:2]: move named import to bottom of @mui

File: src/Components/GuidePanel.jsx [hunk 19]
```diff
@@ -1,9 +1,61 @@
...
+import IconButton from '@mui/material/IconButton'
 import Paper from '@mui/material/Paper'
 import Close from '../assets/3D/clear.svg'
 import Question from '../assets/3D/help.svg'
 
+/**
+ * A UI control to toggle Guide panel on and off
+ * @param {Number} offsetTop position of the panel
+ * @param {Object} viewer IFC viewer.
```
Comment (pablo-mayrgundter) [inline src/Components/GuidePanel.jsx:11]: remove this param

File: src/Components/GuidePanel.jsx [hunk 20]
```diff
@@ -1,9 +1,61 @@
...
+ * @return {Object} The GuidePanelControl react component.
+ */
+export default function GuidePanelControl({offsetTop}) {
+  const [open, setOpen]=React.useState(false)
```
Comment (pablo-mayrgundter) [inline src/Components/GuidePanel.jsx:15]: space around =

File: src/Components/GuidePanel.jsx [hunk 21]
```diff
@@ -1,9 +1,61 @@
...
+    </IconButton>)
+}
+
+
+/**
+ * Guide Panel component
+ * @param {string} offset off set from the top of the page
+ * @param {boolean} openToggle toggle state that is set in the control
+ * @return {Component} componet
+ */
```
Comment (pablo-mayrgundter) [inline src/Components/GuidePanel.jsx:33]:
```
/**
 * Guide Panel component
 * @param {boolean} openToggle React state toggle.
 * @param {string} offset Distance from from the top of the page in css.
 * @return {Object} Guide panel react component.
 */
```

File: src/Components/ItemProperties.jsx [hunk 22]
```diff
@@ -56,11 +56,11 @@ async function createPropertyTable(props, viewer, serial = 0, isPset = false) {
...
                   .filter((key) => !(isPset && (key == 'expressID' || key == 'Name')))
                   .map(
-                    async (key, ndx) => {
-                      const val = props[key];
-                      console.log('createPropertyTable', key, val)
-                      return await prettyProps(key, val, viewer, ndx)
-                    }
+                      async (key, ndx) => {
+                        const val = props[key]
+                        console.log('createPropertyTable', key, val)
```
Comment (pablo-mayrgundter) [inline src/Components/ItemProperties.jsx:61]: remove debug log

File: src/Components/ItemProperties.test.jsx [hunk 23]
```diff
@@ -3,7 +3,6 @@ import {render, screen, waitFor} from '@testing-library/react'
 import {MockViewer, newMockStringValueElt} from '../utils/IfcMock.test'
 import {mockRoutes} from '../BaseRoutesMock.test'
 import ItemProperties from './ItemProperties'
-import testObj from './ItemProperties.testobj.json'
```
Comment (pablo-mayrgundter) [inline src/Components/ItemProperties.test.jsx:6]: Please leave it in as a dep.

File: src/Share.jsx [hunk 24]
```diff
@@ -79,11 +80,11 @@ const theme = useMemo(() => {
...
         },
         tonalOffset: 1,
-        },
-      shape:{
-        borderRadius:10},
-      });
-  }, [mode]);
+      },
+      shape: {
+        borderRadius: 10},
```
Comment (pablo-mayrgundter) [inline src/Share.jsx:85]: bring trailing } to next line

File: src/utils/TreeUtils.js [hunk 25]
```diff
@@ -1,4 +1,9 @@
-function visitTree(elt, observeCb) {
+/**
+ * Recursively visit nodes if the node has children
+ * @param {Object} elt tree element
+ * @param {function} observeCb obeserve function
```
Comment (pablo-mayrgundter) [inline src/utils/TreeUtils.js:4]: @param {function} observeCb Callback for each child/element pair: observeCb(child, elt).

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
Please remove .eslintirgnore since it's empty.  Probably won't need it.

**Review by pablo-mayrgundter — APPROVED** [review]

**Comment by pablo-mayrgundter** [comment]
Looks great!  Welcome to the love of eslint! ;)

**Pull request merged**